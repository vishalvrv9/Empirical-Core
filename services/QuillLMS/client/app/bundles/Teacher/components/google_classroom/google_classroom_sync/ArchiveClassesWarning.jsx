import React from 'react'
import $ from 'jquery'
import pluralize from 'pluralize'

import Modal from 'react-bootstrap/lib/Modal';

export default React.createClass({

    render: function() {
      let archivedCount;
      if (this.props.data && this.props.data.archivedCount) {
       archivedCount = this.props.data.archivedCount;
     } else {
       archivedCount = 0;
     }
        return (
            <Modal {...this.props} show={this.props.show} onHide={this.props.hideModal} dialogClassName='google-classroom-modal archive-warning' restoreFocus={true}>
              <Modal.Body>
                  <img className='pull-right react-bootstrap-close' onClick={this.props.hideModal} src='/images/close_x.svg' alt='close-modal'/>
                  <h1 className='q-h2'><i className="fa fa-exclamation-triangle"></i>Warning: Archiving Classroom</h1>
                  <h3 className='subheader'>You are archiving <strong>{`${archivedCount} ${pluralize('classroom', archivedCount)}.`}</strong></h3>
                  <p>
                    When a classroom is deselected, Quill will archive the classroom and it will no longer sync that classroom with Google Classroom. If you would like to use the classroom, please go back and select it
                  </p>
                  <div className="button-wrapper">
                    <button className="q-button cta-button bg-white text-black close-modal" onClick={this.props.hideModal}>
                      Back
                    </button>
                    <button className="q-button cta-button bg-quillgreen text-white sync-classrooms" onClick={this.props.syncClassroomsAjax}>
                      Sync Classrooms
                    </button>
                  </div>
              </Modal.Body>
            </Modal>
        )
    }

})
