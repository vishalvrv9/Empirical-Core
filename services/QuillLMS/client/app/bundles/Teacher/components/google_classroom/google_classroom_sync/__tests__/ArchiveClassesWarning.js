import React from 'react';
import { shallow } from 'enzyme';
import ArchiveClassesWarning from '../ArchiveClassesWarning.jsx';

describe('the ArchiveClassesWarning component', () => {
  it('should render', () => {
    const wrapper = shallow(<ArchiveClassesWarning />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the number of classes being archived', () => {
    const wrapper = shallow(<ArchiveClassesWarning data={{ archivedCount: 10, }} />);
    expect(wrapper.find('strong').first().text()).toBe('10 classrooms.');
  });

  it('should trigger the hideModal callBack when the back button is clicked', () => {
    const mockBackBtn = jest.fn();
    const wrapper = shallow(<ArchiveClassesWarning hideModal={mockBackBtn} />);
    wrapper.find('.close-modal').first().simulate('click');
    expect(mockBackBtn.mock.calls.length).toBe(1);
  });

  it('should trigger the syncClassroomsAjax callBack when the sync button is clicked', () => {
    const mockBackBtn = jest.fn();
    const wrapper = shallow(<ArchiveClassesWarning syncClassroomsAjax={mockBackBtn} />);
    wrapper.find('.sync-classrooms').first().simulate('click');
    expect(mockBackBtn.mock.calls.length).toBe(1);
  });
});
