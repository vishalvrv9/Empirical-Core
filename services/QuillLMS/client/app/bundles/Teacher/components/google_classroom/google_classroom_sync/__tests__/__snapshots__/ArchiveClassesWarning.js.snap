// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ArchiveClassesWarning component should render 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="google-classroom-modal archive-warning"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "modals": Array [],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <img
      alt="close-modal"
      className="pull-right react-bootstrap-close"
      src="/images/close_x.svg"
    />
    <h1
      className="q-h2"
    >
      <i
        className="fa fa-exclamation-triangle"
      />
      Warning: Archiving Classroom
    </h1>
    <h3
      className="subheader"
    >
      You are archiving 
      <strong>
        0 classrooms.
      </strong>
    </h3>
    <p>
      When a classroom is deselected, Quill will archive the classroom and it will no longer sync that classroom with Google Classroom. If you would like to use the classroom, please go back and select it
    </p>
    <div
      className="button-wrapper"
    >
      <button
        className="q-button cta-button bg-white text-black close-modal"
      >
        Back
      </button>
      <button
        className="q-button cta-button bg-quillgreen text-white sync-classrooms"
      >
        Sync Classrooms
      </button>
    </div>
  </ModalBody>
</Modal>
`;
